File: CommunityWebApp/ClientApp/src/components/restaurant/Restaurant.js
import React from 'react';
import { CardBody, Card } from 'reactstrap';

export default function Restaurant({ restaurant }) {
    if (restaurant) {
        return (
            <Card>
                <CardBody>
                    <div>
                        <h3>{restaurant.businessName}</h3>
                    </div>
                    <div>
                        {restaurant.addressLine1}, {restaurant.addressLine2}, {restaurant.town}, {restaurant.county}, {restaurant.postCode}
                    </div>
                    <div>
                        tel: {restaurant.phone}
                    </div>
                    <div>
                        email: {restaurant.email}
                    </div>
                    <div>
                        Verified? {restaurant.isVerified === true ? "Yes" : "No"}
                    </div>
                    <div>
                        Number of available servings: {restaurant.numberOfAvailableServings}
                    </div>
                </CardBody>
            </Card>
        );
    }
    else {
        return (<div>No restaurants found</div>);
    }
}

File: Restaurants/Entities/Restaurant.cs

using System.ComponentModel.DataAnnotations;

namespace Restaurants.Entities
{
    public class Restaurant
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public long OwnerId { get; set; }

        [Required]
        [StringLength(255)]
        public string BusinessName { get; set; }

        [Required]
        [StringLength(255)]
        public string AddressLine1 { get; set; }

        [StringLength(255)]
        public string AddressLine2 { get; set; }

        [Required]
        [StringLength(35)]
        public string Town { get; set; }

        [StringLength(35)]
        public string County { get; set; }

        [Required]
        [StringLength(8)]
        public string PostCode { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        [Phone]
        public string Phone { get; set; }

        public int NumberOfAvailableServings { get; set; }
    }
}
